gsi/verification/verifiers.py

```diff
@@ -58,85 +58,106 @@ def _fetch_certs(request, certs_url):
 class Verifier(object):
     """Verifies an ID token and returns the decoded token."""
 
-    def __init__(self, id_token, client_ids=None, request=request.Request(),
+    def __init__(self):
+        """
+        Initializes the verifier
+        """
+        pass
+    
+    def _verify_token_payload(self, id_token, client_ids=None, request_object=request.Request(),
             certs_url=_GOOGLE_OAUTH2_CERTS_URL):
         """
+        Verifies the stored ID Token and returns it's payload in a hashmap form.
+        
         Args:
             id_token (Union[str, bytes]): The encoded token.
             
             client_ids (list[str]): List of CLIENT_ID values of all audiences that use this backend. If None,
                 then the audience is not verified.
             
-            request (gsi.transport.Request): The object used to make
+            request_object (gsi.transport.Request): The object used to make
                 HTTP requests.
                 
             certs_url (str): The URL that specifies the certificates to use to
                 verify the token. This URL should return JSON in the format of
                 ``{'key id': 'x509 certificate'}``.
-        """
-        self.id_token = id_token
-        self.request = request
-        self.clients = client_ids
-        self.certs_url = certs_url
-    
-    def _verify_token_payload(self):
-        """
-        Verifies the stored ID Token and returns it's payload in a hashmap form.
 
         Returns:
             Mapping[str, Any]: The decoded token payload.
         """
-        certs = _fetch_certs(self.request, self.certs_url)
-        return jwt.decode(self.id_token, certs=certs, audience=self.clients)
+        certs = _fetch_certs(request_object, certs_url)
+        return jwt.decode(id_token, certs=certs, audience=client_ids)
         
-    def verify_token(self):
+    def verify_token(self, id_token, client_ids=None, request_object=request.Request(),
```

Comment:
I think that's reasonable. It would make all calls to verify_token only require the new id_token which is very simple. If they require different parameters for client_ids or request_object then they can just construct a new Verifier, but as you said I don't believe this should happen for most developers

Reply:
Check out the new class and function signatures. They should satisfy what we discussed here